### PR TITLE
[PHI] Migrate gaussian_random kernel

### PR DESCRIPTION
**Pull request by Silv3S**
### PR types
Others

### PR changes
Others

### Describe
- move gaussian_random_mkldnn to PHI,
- unify OneDNN library name across OneDNN related files.

### CODE REVIEW

**Review by piotrekobi — DISMISSED** [review]
LGTM

**Comment by Silv3S** [comment]
@piotrekobi please review

**Comment by jczaja** [comment]
@chenwhql Please review

**Comment by yaomichael** [comment]
@chenwhql pls review.

**Review by jczaja — DISMISSED** [review]

File: paddle/phi/backends/onednn/onednn_helper.h [hunk 1]
```diff
@@ -25,47 +25,75 @@
...
+    case 6:
+      return dnnl::memory::format_tag::abcdef;
+    case 7:
+      return dnnl::memory::format_tag::abcdefg;
+    case 8:
+      return dnnl::memory::format_tag::abcdefgh;
+    case 9:
+      return dnnl::memory::format_tag::abcdefghi;
+    default:
+      PADDLE_THROW(paddle::platform::errors::Unimplemented(
```
Comment (chenwhql) [inline paddle/phi/backends/onednn/onednn_helper.h]: we recommend to use `phi::errors::Unimplemented`

**Review by chenwhql — DISMISSED** [review]
LGTM

**Review by jczaja — APPROVED** [review]

**Pull request merged**